packages/foam-vscode/src/features/create-from-template.ts

```diff
@@ -383,12 +452,21 @@ async function createNoteFromTemplate(
     .readFile(templateUri)
     .then(bytes => bytes.toString());
 
+  const selectedContent = findSelectionContent();
+
   let resolvedValues, templateWithResolvedVariables;
   try {
     [
       resolvedValues,
       templateWithResolvedVariables,
-    ] = await resolveFoamTemplateVariables(templateText);
+    ] = await resolveFoamTemplateVariables(
+      templateText,
+      new Set(),
+      new Map().set(
+        'FOAM_SELECTION_CONTENT',
+        selectedContent !== undefined ? selectedContent.content : ''
```

Comment:
minor: here we could use `selectedContent?.content ?? ''` which is the short version of what you wrote